[Bryan Tee Pak Hong] iP

SpdPnd98 opened the pull request:
Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is FREE!

Features:

 - [X] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

SpdPnd98 committed:
Adding basic skeleton for iP

SpdPnd98 committed:
Added Add feature

SpdPnd98 committed:
Added Task abstraction

SpdPnd98 committed:
Added Deadline, Event, and Todo

SpdPnd98 committed:
Added error handling

SpdPnd98 committed:
Fixes to streamline error logic for level-4

SpdPnd98 committed:
Removing stray comment

SpdPnd98 committed:
updated TextUiTesting

SpdPnd98 committed:
Added exception class

SpdPnd98 committed:
Fix chatbot enums

SpdPnd98 committed:
Adding exception handling for duke

SpdPnd98 committed:
Rename to DukeArgumentException for clarity

SpdPnd98 committed:
Changing expected to reflect new test cases

SpdPnd98 committed:
Added delete function

SpdPnd98 committed:
Fix done

SpdPnd98 committed:
Merge pull request #1 from SpdPnd98/Level-6

Merging from previous week contents

SpdPnd98 committed:
Add save to file features

SpdPnd98 committed:
Add resource folder

SpdPnd98 committed:
Merge pull request #2 from SpdPnd98/branch-Level-7

Mergin level 7

SpdPnd98 committed:
Add date validation for Deadline and Event

SpdPnd98 committed:
Merge pull request #3 from SpdPnd98/branch-Level-8

Add date validation for Deadline and Event

SpdPnd98 committed:
Add more OOP

SpdPnd98 committed:
Merge pull request #4 from SpdPnd98/branch-A-MoreOOP

Add more OOP

SpdPnd98 committed:
Add more OOP

SpdPnd98 committed:
Merge pull request #6 from SpdPnd98/branch-A-packages

Add Packages

SpdPnd98 committed:
Remove stray files

SpdPnd98 committed:
Remove stray files

SpdPnd98 committed:
Add Test skeleton

SpdPnd98 committed:
Add Chatbot Docs

clyveycui commented on src/main/java/duke/Chatbot.java:
```diff
@@ -0,0 +1,126 @@
+            default:
+                this.ui.showNotSupported();
+                return ChatContinue.END;
+        }
```
I think the indentation for switch case here might be off.

clyveycui commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+            this.date = new SimpleDateFormat("dd/mm/yyyy").parse(commandAndDate[1]);
+        } catch (ParseException e) {
+            throw new DukeDateParseException(e);
+        }
```
Interesting design here, putting the parsing of user input in each individual command class.

clyveycui commented on src/main/java/duke/FileDB.java:
```diff
@@ -0,0 +1,52 @@
+import java.io.IOException;
+
+public class FileDB {
+    public static String DEFAULT_SAVE = "ip/src/main/resources/storage.txt";
```
I think this should be lowercase?

SpdPnd98 replied:
This should be all caps and separated with underscores because the default save location should be constant

clyveycui commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,39 @@
+    protected static final String COMPLETED = "[X]";
+    protected static final String NOT_COMPLETED = "[ ]";
+
+    protected final String description;
```
This might be better if it is in uppercase.

nujiak replied:
I think lowercase is fine since description is not a constant

clyveycui commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    protected Chatbot.ChatContinue addTask(Chatbot.TaskCommands command, String input, Ui ui, FileDB fileDB) throws DukeIOException, DukeDateParseException {
+        switch (command) {
+            case DONE:
+                return this.markDone(input, ui);
+            case DELETE:
+                return this.delete(input, ui);
+            case TODO:
+                return this.addTodo(input, fileDB, ui);
+            case DEADLINE:
+                return this.addDeadline(input, fileDB, ui);
+            case EVENT:
+                return this.addEvent(input, fileDB, ui);
+            default:
+                ui.showNotSupported();
+                return Chatbot.ChatContinue.CONTINUE;
+        }
```
Same as mentioned above, the indentation for switch case here might be off.

clyveycui reviewed (commented):
Good job! Looks good to me, just a few minor fixes.

SpdPnd98 committed:
Add JUnit test

SpdPnd98 committed:
Add JavaDocs to most of the classes

SpdPnd98 committed:
Correct style violations

SpdPnd98 committed:
Add find feature

SpdPnd98 committed:
Merge pull request #7 from SpdPnd98/branch-Level-8

Branch level 8

SpdPnd98 committed:
Merge pull request #8 from SpdPnd98/branch-A-MoreOOP

Branch a more oop

SpdPnd98 committed:
Merge pull request #9 from SpdPnd98/branch-A-packages

Add Packages

SpdPnd98 committed:
Merge pull request #10 from SpdPnd98/branch-A-JUnit

Branch j unit

SpdPnd98 committed:
Merge pull request #11 from SpdPnd98/branch-A-JavaDoc

Add java doc

SpdPnd98 committed:
Merge branch 'branch-Level-7' into branch-A-CodingStandard

SpdPnd98 committed:
Merge pull request #12 from SpdPnd98/branch-A-CodingStandard

Correct style violations

SpdPnd98 committed:
Merge branch 'branch-Level-7' into branch-Level-9

SpdPnd98 committed:
Merge pull request #13 from SpdPnd98/branch-Level-9

Add find feature

SpdPnd98 committed:
Merge pull request #14 from SpdPnd98/branch-Level-7

Merge branches from week 3

SpdPnd98 committed:
Add gradle builds

SpdPnd98 committed:
Add checkstyle

SpdPnd98 committed:
Add GUI

SpdPnd98 committed:
Add Duke welcome message

SpdPnd98 committed:
Add file loading and close application function

SpdPnd98 committed:
Add assertions for key steps of programme.

There are currently no checks for correctness of the program.

Assertions need to be  setup to ensure the correctness of the code at keypoints in the chatbot,
this includes the the Chatbot package as well as the implementation of the Chatbot class.

The following has been done to achieve this:
* check the chatting status, ensure user input handlers are not triggered when chatbot is inactive.
* check the output stream was modified after utilizing the guiInterpret, failure of switch back
is potentially application breaking
* check the loading of tasks is done at the beginning, ensure task lists start from a clean slate.

Java's assert is used so that application can ignore such assertions after deployed.

SpdPnd98 committed:
Merge pull request #16 from SpdPnd98/branch-A-Assertions

Branch assertions

SpdPnd98 committed:
Add docs and re-organized codes

SpdPnd98 committed:
Add CI

SpdPnd98 committed:
Add missing Gradle Wrapper JARs

SpdPnd98 committed:
Add Duplicate check

SpdPnd98 committed:
Impoved JavaDocs

SpdPnd98 committed:
Merge pull request #17 from SpdPnd98/branch-A-CI

Add CI

SpdPnd98 committed:
Merge pull request #18 from SpdPnd98/branch-B-DetectDuplicate

Detect Duplicate Feature

SpdPnd98 committed:
Fix style violations

SpdPnd98 committed:
Merge branch 'master' into branch-A-CodeQuality

SpdPnd98 committed:
Merge pull request #19 from SpdPnd98/branch-A-CodeQuality

Add docs and re-organized codes

SpdPnd98 committed:
Set theme jekyll-theme-modernist

SpdPnd98 committed:
Add Ui.png for submission

SpdPnd98 committed:
Modify running class for build file

SpdPnd98 committed:
Fix regression

SpdPnd98 committed:
Add docs

SpdPnd98 committed:
Fix display error

SpdPnd98 committed:
Fix error message display

SpdPnd98 committed:
Fix clipped Ui

SpdPnd98 committed:
Fix regressions
